src/main/scripts/src/app/shared/perspectives/documents/post-it/post-it-documents-perspective.component.ts

```diff
@@ -37,46 +39,70 @@ import {Document} from '../../../../core/dto/document';
     trigger('appear', [
       transition(':enter', [
         style({transform: 'scale(0)'}),
-        animate('0.25s ease-out', style({transform: 'scale(1)'})),
+        animate('0.25s ease-out', style({transform: 'scale(1)'}))
       ]),
       transition(':leave', [
         style({transform: 'scale(1)'}),
-        animate('0.25s ease-out', style({transform: 'scale(0)'})),
+        animate('0.25s ease-out', style({transform: 'scale(0)'}))
       ])
     ])
   ]
 })
 export class PostItDocumentsPerspectiveComponent implements Perspective, OnInit {
 
+  public readonly PERSPECTIVE_SEE_MORE_ADDED_HEIGHT = 400;
+
+  public readonly PERSPECTIVE_OVERFLOW_BASE_HEIGHT = 200;
+
+  public readonly PERSPECTIVE_BASE_HEIGHT = 450;
+
+  public readonly MASONRY_GRID = 'grid';
+
   @Input()
   public query: string;
 
   @Input()
   public editable: boolean;
 
   @Input()
-  public height: string = '500px';
+  public height = this.PERSPECTIVE_BASE_HEIGHT;
 
   public collection: Collection;
 
-  public addDocumentAttribute: string;
-  public addButtonColor: string;
-  public cursorOnAddButton: boolean;
+  public attributes: Attribute[] = [];
+
+  public documents: Document[] = [];
+
+  /**
+   * To prevent sending data after each data change, the timer provides 'buffering', by waiting a while after each change before sending
+   */
+  private updateTimer: number;
+
+  private sendDocumentUpdate: () => void;
+
+  private updatePending: boolean;
 
-  public documents: Document[];
+  private previouslyEditedDocument: Document;
+
+  private layout: any;
 
   constructor(private documentService: DocumentService,
               private collectionService: CollectionService,
               private workspaceService: WorkspaceService,
               private route: ActivatedRoute) {
   }
 
+  public generateId(index: number) {
+    return `Document${index}`;
+  }
+
   public ngOnInit() {
     this.initializeWorkspace();
-    this.fetchDocuments();
+    this.fetchCollections();
+    this.fetchAttributes();
   }
```

Comment:
It's better to read data in one stream. You can use `combineLatest` observable to achieve it.
https://www.learnrxjs.io/operators/combination/combinelatest.html

Maybe it's not necessary to implement it this way now, but keep in mind that in future we will be optimizing each line of code.